tests/Tokenizer/Transformer/NamedArgumentTransformerTest.php

```diff
@@ -64,8 +64,7 @@ public function a($foo){}
                 }
 
                 $foo = new Bar();
-                $foo->a(foo: 1);
-            ',
+                $foo->a(foo: 1);',
```

Comment:
I find it handy that `'` is on next line.
whe I want to modify the code (Eg by adding new line), I do not need to move `',` (basically all the same reasoning as behind trailing `,` in arrays etc

Reply:
This is pure preparation PR for multiline to heredoc conversion, closing marker will be then on a separate line,.

Reply:
if only purpose of this PR is to have another PR, both with massive diff, and no benefit of this PR itself, i suggest to skip this PR and focus on 7706 only.

if 7706 is having logic + CS change, split that into 2. if only CS change, ok to have one.

Reply:
@keradus [`+74,844 −58,592`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7706#issuecomment-1884089162), I'm out 😉. If you want to review such monstrocity at once, then 👍.

Reply:
@Wirone that is the reason of this PR. In #7706, we can they apply only a fixer changes which you can verify by local rerun or even compare if all AST strings are effectively the same :)

Reply:
To what I understood, this PR is pre-head for 2nd PR, but both touching same lines? just step by step (move new line, move to new syntax).

Overall, I fail to see benefit of this PR as-is, and I'm struggling with 2nd PR too.
There is huge diff, causing conflicts for multiple PRs, while I don't see actual value to introduce it.

Reply:
both PR changes same text, eg
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7709/files#diff-a7c7f042383b8fe9dfb601cfd7521e5523cb196b3a43d04ce245fdfa75bdf413L78
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7706/files#diff-a7c7f042383b8fe9dfb601cfd7521e5523cb196b3a43d04ce245fdfa75bdf413L78

if we do massive CS change for same code lines, it should be one PR, 2 commits - not 2 PRs.

(I'm still not convinced for value behind this massive change, OK to be convinced with ppl supporting the move)

Reply:
I see the value in #7706 as this make code intended better:

<img width="752" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/42154c73-7c22-47fc-8b4f-892dbd08a154">

Since whitespace is trimmed on each line (like in ending marker), the outcome is the same, while visually it makes strings to be closer to where they're defined. Obviously I did not look at whole PR and I agree with @julienfalque's [comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7706/files#r1447075153), but in general it's improvement from my perspective.

Reply:
The value is described in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7665 fixer introduction - multiline strings breaks indentation visually, the solution is heredoc/nowdoc syntax. This is what CS/CS fixes are about.

Reply:
> if we do massive CS change for same code lines, it should be one PR, 2 commits - not 2 PRs.

Actually I can agree with this, but it would really be helpful with changes introduced as separate commits, describing the process. To make review easier maybe we can split PR into `src` and `tests`?

Reply:
PR can be either squashed, rebased on master, or merged. In both 2 later cases, the landed commits must match the PR commits.

As it is good practice to not commit (large) changes done by fixer together with config changes, we need 2 PRs. So let's please agree on 2 separate PRs and continual improvement :)

Reply:
Well, it all depends what is considered as a "single change" from repo's perspective. If it's "changing multiline string to heredoc", then it does not matter that multiple steps were required to achieve it. It's @keradus' decision I believe.

Reply:
While I do see the value in standardizing those multiline strings accross the project, I also see the cost of doing this change and I'm really not sure it's worth it. I mean, sure, a lot of strings, especially in tests, are not correctly indented, but is it such a major problem? And if yes, does rewriting all these strings remove completely the problem such as changing the code will be much easier? I don't think so.

IMO the goal is legit, but I would perfer making a lot of small steps (e.g. rewrite a string when we need to change its value or some related code) rather than review the massive #7706. Even in this PR, I see strings were we need to introduce some concatenations with `"\n"`s simply to avoid having actual newlines, which kind of defeats the point of improving code readability.